scripts/devops_tasks/resolve_installer_location.py

```diff
@@ -0,0 +1,158 @@
+                )
+                exit(1)
+            time.sleep(10)
+
+
+def install_selected_python_version(installer_url, installer_folder):
+    current_plat = platform.system().lower()
+
+    installer_folder = os.path.normpath(os.path.abspath(installer_folder))
+    if not os.path.exists(installer_folder):
+        os.mkdir(installer_folder)
+    local_installer_ref = os.path.join(
+        installer_folder,
+        "local" + (".zip" if installer_folder.endswith("zip") else ".tar.gz"),
+    )
+
+    download_installer(installer_url, local_installer_ref)
+
+    if current_plat == "windows":
+        with zipfile.ZipFile(local_installer_ref, "r") as zip_file:
+            zip_file.extractall(installer_folder)
+        try:
+            check_call(WIN_INSTALL_ARRAY, cwd=installer_folder)
+        except CalledProcessError as err:
+            print(err)
+            exit(1)
+
+    else:
+        with tarfile.open(local_installer_ref) as tar_file:
+            tar_file.extractall(installer_folder)
+        try:
+            check_call(UNIX_INSTALL_ARRAY, cwd=installer_folder)
+        except CalledProcessError as err:
+            print(err)
+            exit(1)
+
+
+def get_installer_url(requested_version, version_manifest):
+    current_plat = platform.system().lower()
+
+    print("Current Platform Is {}".format(platform.platform()))
+
+    if version_manifest[requested_version]:
+        found_installers = version_manifest[requested_version]["files"]
+
+        # filter anything that's not x64. we don't care.
+        x64_installers = [
+            file_def for file_def in found_installers if file_def["arch"] == "x64"
+        ]
+
+        if current_plat == "windows":
+            return [
+                installer
+                for installer in x64_installers
+                if installer["platform"] == "win32"
+            ][0]
+        elif current_plat == "darwin":
+            return [
+                installer
+                for installer in x64_installers
+                if installer["platform"] == current_plat
+            ][0]
+        else:
+            return [
+                installer
+                for installer in x64_installers
+                if installer["platform"] == "linux"
+                and installer["platform_version"] == CURRENT_UBUNTU_VERSION
+            ][0]
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Takes the incoming request and attempts to resolve the version spec."
+    )
+
+    parser.add_argument(
+        "version_spec",
+        nargs="?",
+        help=("The version specifier passed in to the UsePythonVersion extended task."),
+    )
+
+    parser.add_argument(
+        "--installer_folder",
+        dest="installer_folder",
+        help=(
+            "The folder where the found installer will be extracted into and run from."
+        ),
+    )
+
+    args = parser.parse_args()
+    max_precached_version = Version("3.8.6")
```

Comment:
Do we have any plans to update this? Does it need to be updated?

Reply:
Yes. Will need to update this once the next major version of python comes out. Once a year. Given that we'll be on gh-actions eventually (and we'll toss this entire thing), I'm betting we'll do that exactly once.